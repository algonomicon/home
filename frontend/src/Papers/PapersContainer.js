import { Component } from 'inferno'
import { createComponent } from 'inferno-fela'
import CategoriesContainer from '../Components/Categories/CategoriesContainer'
import ShowcaseContainer from './Showcase/ShowcaseContainer'
import FiltersContainer from './Filters/FiltersContainer'

const styles = {
  wrapper: () => ({
    maxWidth: '1024px',
    margin: '0 auto',
    padding: '1.4rem'
  }),
  grid: () => ({
    display: 'grid',
    gridTemplateColumns: 'minmax(0, 2fr) minmax(0, 1fr)',
    gridTemplateAreas: 
      `'nav nav'
       'main sidebar'`
  }),
  nav: () => ({
    gridArea: 'nav'
  }),
  main: () => ({
    gridArea: 'main',
    borderRight: '1px solid #ccc',
    paddingRight: '1.4rem'
  }),
  sidebar: () => ({
    gridArea: 'sidebar',
    paddingLeft: '1.4rem',
    maxHeight: '100%',
    overflow: 'hidden'
  })
}

const Wrapper = createComponent(styles.wrapper)
const Grid = createComponent(styles.grid)
const Nav = createComponent(styles.nav)
const Main = createComponent(styles.main)
const Sidebar = createComponent(styles.sidebar)

class PapersContainer extends Component {
  render() {
    return (
      <Wrapper>
        <Grid>
          <Nav>
            <CategoriesContainer />
          </Nav>
          <Main>
            <ShowcaseContainer />
          </Main>
          <Sidebar>
            <FiltersContainer />
          </Sidebar>
        </Grid>
      </Wrapper>
    )
  }
}

export default PapersContainer