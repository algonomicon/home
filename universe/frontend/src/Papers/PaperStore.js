import { action, observable } from 'mobx'

class PaperStore {
  @observable papers = []

  @action fetchPapers() {
    fetch('http://localhost:8000/papers')
      .then(response => response.json())
      .then(papers => this.papers = papers)
  }
}

export default PaperStore