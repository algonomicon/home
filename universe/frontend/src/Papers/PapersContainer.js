import { Component } from 'inferno'
import { createComponent } from 'inferno-fela'
import CategoriesContainer from '../Components/Categories/CategoriesContainer'
import ShowcaseContainer from './Showcase/ShowcaseContainer'
import FiltersContainer from './Filters/FiltersContainer'

const styles = {
  wrapper: () => ({
    maxWidth: '1024px',
    margin: '0 auto',
    padding: '1.4rem'
  }),
  grid: () => ({
    display: 'grid',
    gridTemplateColumns: 'minmax(0, 2fr) minmax(0, 1fr)',
    gridTemplateAreas: 
      `'nav nav'
       'main sidebar'`
  }),
  nav: () => ({
    gridArea: 'nav'
  }),
  main: () => ({
    gridArea: 'main',
    borderRight: '1px solid #ccc',
    paddingRight: '1.4rem'
  }),
  sidebar: () => ({
    gridArea: 'sidebar',
    paddingLeft: '1.4rem',
    maxHeight: '100%',
    overflow: 'hidden'
  }),
  heading: () => ({
    color: '#333',
    borderBottom: '1px solid #ccc',
    fontSize: '1.4rem',
    padding: '1.4rem 0',
    lineHeight: '1',
    fontFamily: 'monospace',
    display: 'block',
    textTransform: 'uppercase'
  })
}

const Wrapper = createComponent(styles.wrapper)
const Grid = createComponent(styles.grid)
const Nav = createComponent(styles.nav)
const Main = createComponent(styles.main)
const Sidebar = createComponent(styles.sidebar)
const Heading = createComponent(styles.heading, 'h3')

class PapersContainer extends Component {
  render() {
    return (
      <Wrapper>
        <Grid>
          <Nav>
            <CategoriesContainer />
          </Nav>
          <Main>
            <ShowcaseContainer />
          </Main>
          <Sidebar>
            <Heading>Filters</Heading>
            <FiltersContainer />
          </Sidebar>
        </Grid>
      </Wrapper>
    )
  }
}

export default PapersContainer