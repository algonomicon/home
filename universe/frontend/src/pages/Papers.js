import { Component } from 'inferno'
import Categories from '../components/Categories'

class Papers extends Component {
  render() {
    return (
      <div>
        <Categories />
        <h1>Papers</h1>
      </div>
    )
  }
}

export default Papers