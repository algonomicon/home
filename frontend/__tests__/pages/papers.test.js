import React from "react"
import renderer from "react-test-renderer"

import Papers from "../../src/pages/papers"

describe("Papers", () => {
  it("renders correctly", () => {
    const data = {
      "papers": {
        "edges": [
          {
            "node": {
              "title": "A Neural Algorithm of Artistic Style",
              "slug": {
                "current": "a-neural-algorithm-of-artistic-style"
              },
              "author": "Leon A. Gatys, Alexander S. Ecker, Matthias Bethge",
              "abstract": "In fine art, especially painting, humans have mastered the skill to create unique visual experiences through composing a complex interplay between the content and style of an image. Thus far the algorithmic basis of this process is unknown and there exists no artificial system with similar capabilities. However, in other key areas of visual perception such as object and face recognition near-human performance was recently demonstrated by a class of biologically inspired vision models called Deep Neural Networks. Here we introduce an artificial system based on a Deep Neural Network that creates artistic images of high perceptual quality. The system uses neural representations to separate and recombine content and style of arbitrary images, providing a neural algorithm for the creation of artistic images. Moreover, in light of the striking similarities between performance-optimised artificial neural networks and biological vision, our work offers a path forward to an algorithmic understanding of how humans create and perceive artistic imagery."
            }
          }
        ]
      }
    }
    
    const tree = renderer.create(<Papers data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
