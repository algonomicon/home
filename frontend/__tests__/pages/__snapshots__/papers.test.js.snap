// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Papers renders correctly 1`] = `
<div>
  <div
    className="sc-gZMcBi fkYhnm"
  >
    <div
      className="sc-htoDjs exnspz"
    >
      <div
        className="sc-gqjmRU ctxItp"
      >
        <div
          className="logo"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="glitch"
            href="/"
          >
            <svg
              className="glitch__img"
              height="31.75mm"
              version="1.1"
              viewBox="0 0 1170 112.5"
              width="330.2mm"
            >
              <g
                id="layer1"
                transform="translate(962.86 -165.58)"
              >
                <g
                  id="text3336"
                >
                  <path
                    d="m-940.36 188.08v-22.5h45v22.5h22.5v90h-22.5v-22.5h-45v22.5h-22.5v-90h22.5zm45 45v-44.82h-45v44.82h45z"
                    id="path3364"
                  />
                  <path
                    d="m-782.86 278.08v-22.5h-45v-90h-22.5v112.5h67.5z"
                    id="path3366"
                  />
                  <path
                    d="m-670.36 188.08v-22.5h-67.5v22.5h67.5zm-67.5 0h-22.5v67.5h22.5v-67.5zm67.5 90h-67.5v-22.5h45v-22.5h-22.5v-22.5h45v67.5z"
                    id="path3368"
                  />
                  <path
                    d="m-580.36 255.4v-67.14h-45v67.14h45zm-45 0.18h-22.5v-67.5h22.5v-22.5h45v22.5h22.5v67.5h-22.5v22.5h-45v-22.5z"
                    id="path3370"
                  />
                  <path
                    d="m-467.86 233.08v45h22.5v-112.5h-22.5v45h-22.5v22.5h22.5zm-45-22.5h22.5v-22.5h-22.5v-22.5h-22.5v112.5h22.5v-67.5z"
                    id="path3372"
                  />
                  <path
                    d="m-355.36 255.4v-67.14h-45v67.14h45zm-45 0.18h-22.5v-67.5h22.5v-22.5h45v22.5h22.5v67.5h-22.5v22.5h-45v-22.5z"
                    id="path3374"
                  />
                  <path
                    d="m-287.86 188.08v-22.5h-22.5v112.5h22.5v-67.5h22.5v-22.5h-22.5zm22.5 45h22.5v-22.5h-22.5v22.5zm45-45h-22.5v22.5h22.5v67.5h22.5v-112.5h-22.5v22.5z"
                    id="path3376"
                  />
                  <path
                    d="m-130.36 188.08h22.5v-22.5h-67.5v22.5h22.5v67.5h-22.5v22.5h67.5v-22.5h-22.5v-67.5z"
                    id="path3378"
                  />
                  <path
                    d="m-17.857 188.08v-22.5h-45v22.5h45zm-45 0h-22.5v67.5h22.5v-67.5zm45 90v-22.5h-45v22.5h45z"
                    id="path3380"
                  />
                  <path
                    d="m72.143 255.4v-67.14h-45v67.14h45zm-45 0.18h-22.5v-67.5h22.5v-22.5h45v22.5h22.5v67.5h-22.5v22.5h-45v-22.5z"
                    id="path3382"
                  />
                  <path
                    d="m184.64 233.08v45h22.5v-112.5h-22.5v45h-22.5v22.5h22.5zm-45-22.5h22.5v-22.5h-22.5v-22.5h-22.5v112.5h22.5v-67.5z"
                    id="path3384"
                  />
                </g>
              </g>
            </svg>
            <svg
              className="glitch__img"
              height="31.75mm"
              version="1.1"
              viewBox="0 0 1170 112.5"
              width="330.2mm"
            >
              <g
                id="layer1"
                transform="translate(962.86 -165.58)"
              >
                <g
                  id="text3336"
                >
                  <path
                    d="m-940.36 188.08v-22.5h45v22.5h22.5v90h-22.5v-22.5h-45v22.5h-22.5v-90h22.5zm45 45v-44.82h-45v44.82h45z"
                    id="path3364"
                  />
                  <path
                    d="m-782.86 278.08v-22.5h-45v-90h-22.5v112.5h67.5z"
                    id="path3366"
                  />
                  <path
                    d="m-670.36 188.08v-22.5h-67.5v22.5h67.5zm-67.5 0h-22.5v67.5h22.5v-67.5zm67.5 90h-67.5v-22.5h45v-22.5h-22.5v-22.5h45v67.5z"
                    id="path3368"
                  />
                  <path
                    d="m-580.36 255.4v-67.14h-45v67.14h45zm-45 0.18h-22.5v-67.5h22.5v-22.5h45v22.5h22.5v67.5h-22.5v22.5h-45v-22.5z"
                    id="path3370"
                  />
                  <path
                    d="m-467.86 233.08v45h22.5v-112.5h-22.5v45h-22.5v22.5h22.5zm-45-22.5h22.5v-22.5h-22.5v-22.5h-22.5v112.5h22.5v-67.5z"
                    id="path3372"
                  />
                  <path
                    d="m-355.36 255.4v-67.14h-45v67.14h45zm-45 0.18h-22.5v-67.5h22.5v-22.5h45v22.5h22.5v67.5h-22.5v22.5h-45v-22.5z"
                    id="path3374"
                  />
                  <path
                    d="m-287.86 188.08v-22.5h-22.5v112.5h22.5v-67.5h22.5v-22.5h-22.5zm22.5 45h22.5v-22.5h-22.5v22.5zm45-45h-22.5v22.5h22.5v67.5h22.5v-112.5h-22.5v22.5z"
                    id="path3376"
                  />
                  <path
                    d="m-130.36 188.08h22.5v-22.5h-67.5v22.5h22.5v67.5h-22.5v22.5h67.5v-22.5h-22.5v-67.5z"
                    id="path3378"
                  />
                  <path
                    d="m-17.857 188.08v-22.5h-45v22.5h45zm-45 0h-22.5v67.5h22.5v-67.5zm45 90v-22.5h-45v22.5h45z"
                    id="path3380"
                  />
                  <path
                    d="m72.143 255.4v-67.14h-45v67.14h45zm-45 0.18h-22.5v-67.5h22.5v-22.5h45v22.5h22.5v67.5h-22.5v22.5h-45v-22.5z"
                    id="path3382"
                  />
                  <path
                    d="m184.64 233.08v45h22.5v-112.5h-22.5v45h-22.5v22.5h22.5zm-45-22.5h22.5v-22.5h-22.5v-22.5h-22.5v112.5h22.5v-67.5z"
                    id="path3384"
                  />
                </g>
              </g>
            </svg>
            <svg
              className="glitch__img"
              height="31.75mm"
              version="1.1"
              viewBox="0 0 1170 112.5"
              width="330.2mm"
            >
              <g
                id="layer1"
                transform="translate(962.86 -165.58)"
              >
                <g
                  id="text3336"
                >
                  <path
                    d="m-940.36 188.08v-22.5h45v22.5h22.5v90h-22.5v-22.5h-45v22.5h-22.5v-90h22.5zm45 45v-44.82h-45v44.82h45z"
                    id="path3364"
                  />
                  <path
                    d="m-782.86 278.08v-22.5h-45v-90h-22.5v112.5h67.5z"
                    id="path3366"
                  />
                  <path
                    d="m-670.36 188.08v-22.5h-67.5v22.5h67.5zm-67.5 0h-22.5v67.5h22.5v-67.5zm67.5 90h-67.5v-22.5h45v-22.5h-22.5v-22.5h45v67.5z"
                    id="path3368"
                  />
                  <path
                    d="m-580.36 255.4v-67.14h-45v67.14h45zm-45 0.18h-22.5v-67.5h22.5v-22.5h45v22.5h22.5v67.5h-22.5v22.5h-45v-22.5z"
                    id="path3370"
                  />
                  <path
                    d="m-467.86 233.08v45h22.5v-112.5h-22.5v45h-22.5v22.5h22.5zm-45-22.5h22.5v-22.5h-22.5v-22.5h-22.5v112.5h22.5v-67.5z"
                    id="path3372"
                  />
                  <path
                    d="m-355.36 255.4v-67.14h-45v67.14h45zm-45 0.18h-22.5v-67.5h22.5v-22.5h45v22.5h22.5v67.5h-22.5v22.5h-45v-22.5z"
                    id="path3374"
                  />
                  <path
                    d="m-287.86 188.08v-22.5h-22.5v112.5h22.5v-67.5h22.5v-22.5h-22.5zm22.5 45h22.5v-22.5h-22.5v22.5zm45-45h-22.5v22.5h22.5v67.5h22.5v-112.5h-22.5v22.5z"
                    id="path3376"
                  />
                  <path
                    d="m-130.36 188.08h22.5v-22.5h-67.5v22.5h22.5v67.5h-22.5v22.5h67.5v-22.5h-22.5v-67.5z"
                    id="path3378"
                  />
                  <path
                    d="m-17.857 188.08v-22.5h-45v22.5h45zm-45 0h-22.5v67.5h22.5v-67.5zm45 90v-22.5h-45v22.5h45z"
                    id="path3380"
                  />
                  <path
                    d="m72.143 255.4v-67.14h-45v67.14h45zm-45 0.18h-22.5v-67.5h22.5v-22.5h45v22.5h22.5v67.5h-22.5v22.5h-45v-22.5z"
                    id="path3382"
                  />
                  <path
                    d="m184.64 233.08v45h22.5v-112.5h-22.5v45h-22.5v22.5h22.5zm-45-22.5h22.5v-22.5h-22.5v-22.5h-22.5v112.5h22.5v-67.5z"
                    id="path3384"
                  />
                </g>
              </g>
            </svg>
            <svg
              className="glitch__img"
              height="31.75mm"
              version="1.1"
              viewBox="0 0 1170 112.5"
              width="330.2mm"
            >
              <g
                id="layer1"
                transform="translate(962.86 -165.58)"
              >
                <g
                  id="text3336"
                >
                  <path
                    d="m-940.36 188.08v-22.5h45v22.5h22.5v90h-22.5v-22.5h-45v22.5h-22.5v-90h22.5zm45 45v-44.82h-45v44.82h45z"
                    id="path3364"
                  />
                  <path
                    d="m-782.86 278.08v-22.5h-45v-90h-22.5v112.5h67.5z"
                    id="path3366"
                  />
                  <path
                    d="m-670.36 188.08v-22.5h-67.5v22.5h67.5zm-67.5 0h-22.5v67.5h22.5v-67.5zm67.5 90h-67.5v-22.5h45v-22.5h-22.5v-22.5h45v67.5z"
                    id="path3368"
                  />
                  <path
                    d="m-580.36 255.4v-67.14h-45v67.14h45zm-45 0.18h-22.5v-67.5h22.5v-22.5h45v22.5h22.5v67.5h-22.5v22.5h-45v-22.5z"
                    id="path3370"
                  />
                  <path
                    d="m-467.86 233.08v45h22.5v-112.5h-22.5v45h-22.5v22.5h22.5zm-45-22.5h22.5v-22.5h-22.5v-22.5h-22.5v112.5h22.5v-67.5z"
                    id="path3372"
                  />
                  <path
                    d="m-355.36 255.4v-67.14h-45v67.14h45zm-45 0.18h-22.5v-67.5h22.5v-22.5h45v22.5h22.5v67.5h-22.5v22.5h-45v-22.5z"
                    id="path3374"
                  />
                  <path
                    d="m-287.86 188.08v-22.5h-22.5v112.5h22.5v-67.5h22.5v-22.5h-22.5zm22.5 45h22.5v-22.5h-22.5v22.5zm45-45h-22.5v22.5h22.5v67.5h22.5v-112.5h-22.5v22.5z"
                    id="path3376"
                  />
                  <path
                    d="m-130.36 188.08h22.5v-22.5h-67.5v22.5h22.5v67.5h-22.5v22.5h67.5v-22.5h-22.5v-67.5z"
                    id="path3378"
                  />
                  <path
                    d="m-17.857 188.08v-22.5h-45v22.5h45zm-45 0h-22.5v67.5h22.5v-67.5zm45 90v-22.5h-45v22.5h45z"
                    id="path3380"
                  />
                  <path
                    d="m72.143 255.4v-67.14h-45v67.14h45zm-45 0.18h-22.5v-67.5h22.5v-22.5h45v22.5h22.5v67.5h-22.5v22.5h-45v-22.5z"
                    id="path3382"
                  />
                  <path
                    d="m184.64 233.08v45h22.5v-112.5h-22.5v45h-22.5v22.5h22.5zm-45-22.5h22.5v-22.5h-22.5v-22.5h-22.5v112.5h22.5v-67.5z"
                    id="path3384"
                  />
                </g>
              </g>
            </svg>
            <svg
              className="glitch__img"
              height="31.75mm"
              version="1.1"
              viewBox="0 0 1170 112.5"
              width="330.2mm"
            >
              <g
                id="layer1"
                transform="translate(962.86 -165.58)"
              >
                <g
                  id="text3336"
                >
                  <path
                    d="m-940.36 188.08v-22.5h45v22.5h22.5v90h-22.5v-22.5h-45v22.5h-22.5v-90h22.5zm45 45v-44.82h-45v44.82h45z"
                    id="path3364"
                  />
                  <path
                    d="m-782.86 278.08v-22.5h-45v-90h-22.5v112.5h67.5z"
                    id="path3366"
                  />
                  <path
                    d="m-670.36 188.08v-22.5h-67.5v22.5h67.5zm-67.5 0h-22.5v67.5h22.5v-67.5zm67.5 90h-67.5v-22.5h45v-22.5h-22.5v-22.5h45v67.5z"
                    id="path3368"
                  />
                  <path
                    d="m-580.36 255.4v-67.14h-45v67.14h45zm-45 0.18h-22.5v-67.5h22.5v-22.5h45v22.5h22.5v67.5h-22.5v22.5h-45v-22.5z"
                    id="path3370"
                  />
                  <path
                    d="m-467.86 233.08v45h22.5v-112.5h-22.5v45h-22.5v22.5h22.5zm-45-22.5h22.5v-22.5h-22.5v-22.5h-22.5v112.5h22.5v-67.5z"
                    id="path3372"
                  />
                  <path
                    d="m-355.36 255.4v-67.14h-45v67.14h45zm-45 0.18h-22.5v-67.5h22.5v-22.5h45v22.5h22.5v67.5h-22.5v22.5h-45v-22.5z"
                    id="path3374"
                  />
                  <path
                    d="m-287.86 188.08v-22.5h-22.5v112.5h22.5v-67.5h22.5v-22.5h-22.5zm22.5 45h22.5v-22.5h-22.5v22.5zm45-45h-22.5v22.5h22.5v67.5h22.5v-112.5h-22.5v22.5z"
                    id="path3376"
                  />
                  <path
                    d="m-130.36 188.08h22.5v-22.5h-67.5v22.5h22.5v67.5h-22.5v22.5h67.5v-22.5h-22.5v-67.5z"
                    id="path3378"
                  />
                  <path
                    d="m-17.857 188.08v-22.5h-45v22.5h45zm-45 0h-22.5v67.5h22.5v-67.5zm45 90v-22.5h-45v22.5h45z"
                    id="path3380"
                  />
                  <path
                    d="m72.143 255.4v-67.14h-45v67.14h45zm-45 0.18h-22.5v-67.5h22.5v-22.5h45v22.5h22.5v67.5h-22.5v22.5h-45v-22.5z"
                    id="path3382"
                  />
                  <path
                    d="m184.64 233.08v45h22.5v-112.5h-22.5v45h-22.5v22.5h22.5zm-45-22.5h22.5v-22.5h-22.5v-22.5h-22.5v112.5h22.5v-67.5z"
                    id="path3384"
                  />
                </g>
              </g>
            </svg>
          </a>
        </div>
        <div
          className="sc-VigVT cQowiX"
        >
          <a
            className="sc-jTzLTM XXXSU"
            href="/about"
          >
            About
          </a>
          <a
            className="sc-jTzLTM XXXSU"
            href="/contact"
          >
            Contact
          </a>
        </div>
      </div>
      <div
        className="sc-fjdhpX jVgRYA"
      >
        <h1
          className="sc-jzJRlG diYmFi"
        >
          Machine Learning for Fun and Profit
        </h1>
        <h4
          className="sc-cSHVUG dHvNZW"
        >
          Still under construction. Thanks for bearing with us! Sign up for updates.
        </h4>
        <div
          className="sc-kAzzGY iGneOL"
        >
          <div
            className="sc-kgoBCf heGha"
          >
            <div
              className="sc-kGXeez lmfQek"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-user-astronaut fa-w-14 fa-3x"
                data-icon="user-astronaut"
                data-prefix="fas"
                role="img"
                style={
                  Object {
                    "display": "block",
                  }
                }
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M64 224h13.5c24.7 56.5 80.9 96 146.5 96s121.8-39.5 146.5-96H384c8.8 0 16-7.2 16-16v-96c0-8.8-7.2-16-16-16h-13.5C345.8 39.5 289.6 0 224 0S102.2 39.5 77.5 96H64c-8.8 0-16 7.2-16 16v96c0 8.8 7.2 16 16 16zm40-88c0-22.1 21.5-40 48-40h144c26.5 0 48 17.9 48 40v24c0 53-43 96-96 96h-48c-53 0-96-43-96-96v-24zm72 72l12-36 36-12-36-12-12-36-12 36-36 12 36 12 12 36zm151.6 113.4C297.7 340.7 262.2 352 224 352s-73.7-11.3-103.6-30.6C52.9 328.5 0 385 0 454.4v9.6c0 26.5 21.5 48 48 48h80v-64c0-17.7 14.3-32 32-32h128c17.7 0 32 14.3 32 32v64h80c26.5 0 48-21.5 48-48v-9.6c0-69.4-52.9-125.9-120.4-133zM272 448c-8.8 0-16 7.2-16 16s7.2 16 16 16 16-7.2 16-16-7.2-16-16-16zm-96 0c-8.8 0-16 7.2-16 16v48h32v-48c0-8.8-7.2-16-16-16z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <input
              className="sc-kpOJdX botntV"
              placeholder="example@algonomicon.com"
              type="text"
            />
          </div>
          <button
            className="sc-chPdSV hEdLZH"
          >
            Sign Up!
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-dxgOiQ bGYNPR"
  >
    <div
      className="sc-htoDjs exnspz"
    >
      <div
        className="sc-bdVaJa knWkmq"
      >
        <a
          className="sc-bwzfXH FtMCK"
          href="/"
        >
          Home
        </a>
        <a
          className="sc-bwzfXH FtMCK"
          href="/articles"
          partiallyActive={true}
        >
          Articles
        </a>
        <a
          className="sc-bwzfXH FtMCK"
          href="/datasets"
          partiallyActive={true}
        >
          Datasets
        </a>
        <a
          className="sc-bwzfXH FtMCK"
          href="/papers"
          partiallyActive={true}
        >
          Papers
        </a>
        <a
          className="sc-bwzfXH FtMCK"
          href="/projects"
          partiallyActive={true}
        >
          Projects
        </a>
        <a
          className="sc-bwzfXH FtMCK"
          href="/snippets"
          partiallyActive={true}
        >
          Snippets
        </a>
      </div>
      <div
        className="sc-ckVGcZ jecudI"
      >
        <div
          className="sc-jKJlTe xPVez"
        >
          <div
            className="sc-cMljjf gOTAMr"
          >
            <a
              href="/papers/a-neural-algorithm-of-artistic-style"
            >
              <h3>
                A Neural Algorithm of Artistic Style
              </h3>
              <p>
                In fine art, especially painting, humans have mastered the skill to create unique visual experiences through composing a complex interplay between the content and style of an image. Thus far the algorithmic basis of this process is unknown and there exists no artificial system with similar capabilit
                ...
              </p>
            </a>
          </div>
        </div>
        <div
          className="sc-iAyFgw iUHcdr"
        >
          <div>
            <h6
              className="sc-htpNat iacXlz"
            >
              Search
            </h6>
            <div
              className="sc-eNQAEJ cLYxXI"
            >
              <div
                className="sc-hMqMXs fpOYzO"
              >
                <div
                  className="sc-kEYyzF bpyYKe"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-search fa-w-16"
                    data-icon="search"
                    data-prefix="fas"
                    role="img"
                    style={
                      Object {
                        "display": "block",
                      }
                    }
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <input
                  className="sc-kkGfuU fYNrch"
                  placeholder="Articles, Datasets, Research and more..."
                  type="text"
                />
              </div>
            </div>
          </div>
          <div>
            <h6
              className="sc-htpNat iacXlz"
            >
              Sort
            </h6>
            <div
              className="sc-hSdWYo eiXuPs"
            >
              <div
                className="sc-eHgmQL ctFGhs"
              >
                <a
                  className="sc-cvbbAY gidDCq"
                >
                  Recent
                </a>
                <a
                  className="sc-cvbbAY gidDCq"
                >
                  Featured
                </a>
                <a
                  className="sc-cvbbAY gidDCq"
                >
                  Trending
                </a>
                <a
                  className="sc-cvbbAY gidDCq"
                >
                  Favorites
                </a>
              </div>
              <div
                className="sc-eHgmQL ctFGhs"
              >
                <a
                  className="sc-cvbbAY gidDCq"
                >
                  All Time
                </a>
                <a
                  className="sc-cvbbAY gidDCq"
                >
                  Past Year
                </a>
                <a
                  className="sc-cvbbAY gidDCq"
                >
                  Past Month
                </a>
                <a
                  className="sc-cvbbAY gidDCq"
                >
                  Past Week
                </a>
              </div>
            </div>
          </div>
          <div>
            <h6
              className="sc-htpNat iacXlz"
            >
              Tags
            </h6>
            <div
              className="sc-jWBwVP fJqglE"
            >
              <a
                className="sc-brqgnP buTaTv"
                href="/tags/art"
              >
                Art
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                Fun
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                Games
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                General
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                News
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                Language
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                Vision
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                Reinforcment
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                Unsupervised
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                Games
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                General
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                News
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                Language
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                Vision
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                Art
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                Fun
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                Games
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                General
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                News
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                Language
              </a>
              <a
                className="sc-brqgnP buTaTv"
              >
                Vision
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-dnqmqq ipcvRL"
  >
    <div
      className="sc-htoDjs exnspz"
    >
      <p
        className="sc-iwsKbI bYpdxf"
      >
        © Algonomicon LLC 2019
      </p>
    </div>
  </div>
</div>
`;
